modules/@apostrophecms/doc/index.js

```diff
@@ -646,6 +646,11 @@ module.exports = {
           // different from the published, which won't exist yet
           doc.modified = true;
         }
+        if (!doc.visibility) {
+          // If the visibility property has been removed from the schema
+          // (images and files), make sure public queries can still match this type
+          doc.visibility = 'public';
```

Comment:
None of the proper values for `visibility` are falsy, so this is a reasonable test and saves performance and lines of code over grabbing the manager and checking the schema etc.

I did not use a promise event handler because each `await` does impact performance slightly and we don't want to introduce a new delay on every insert for this.